php/libraries/Database.class.inc

```diff
@@ -797,7 +803,6 @@ class Database
 
             // If you get here, we just need to build the new array
             $filteredResult[$row[$uniqueKey]] = $row;
-            unset($filteredResult[$row[$uniqueKey]][$uniqueKey]);
```

Comment:
What was the reason the unset was being used here in the first place? I'm just wondering if its done for any particular reason.

Reply:
@ridz1208?

Reply:
removing this line is fine !

basically when I added this function, the result of the query will be indexed by the key but the key would not be in the result itself.

removing it would still be indexed by key but the key would also still be in the array result

Reply:
Would this cause any problem for existing use of the `pselectWithIndexKey` function? Like having a foreach assuming that the key is NOT included.

Reply:
I went through each use of the function in the code, and to the best of my knowledge it will not cause any problems.